src/components/pages/RoleSignup/Applications/Mentor.js

```diff
@@ -44,27 +47,30 @@ const initialFormValues = {
   other_info: '',
 };
 
-const Mentor = () => {
-  const [formValues, handleChange] = useForms(initialFormValues);
-  const [error, setError] = useState('');
-
+const Mentor = ({ dispatch, error, successPage }) => {
+  const [formValues, handleChange, _, setFormValues] =
+    useForms(initialFormValues);
   const history = useHistory();
 
-  const postNewAccount = async newAccount => {
-    try {
-      await axios.post(
-        `${process.env.REACT_APP_API_URI}/application/new/mentor`,
-        newAccount
-      );
-      history.push('/apply/success');
-    } catch (err) {
-      setError(err);
+  useEffect(() => {
+    axiosWithAuth()
+      .get(`/profile/current_user_profile`)
+      .then(res => {
+        setFormValues({ ...formValues, profile_id: res.data.profile_id });
+      })
+      .catch(err => {
+        console.error(err);
+      });
+
+    if (successPage) {
+      history.pushState(successPage);
+    } else if (error) {
+      console.error(error);
     }
-  };
+  }, [successPage, error]);
```

Comment:
How can we get rid of the linter warning? Do we need to include these dependencies somehow?

Reply:
I can add the history dependency, however adding formValues and setFormValues causes an infinite loop because it will infinitely find and update the profile_id. I am not sure why the linter is complaining about this since it will break it if I do what it is asking.